azkaban-common/src/main/java/azkaban/scheduler/Schedule.java

```diff
@@ -182,11 +217,25 @@ public long getSubmitTime() {
     return submitTime;
   }
 
+  public String getCronExpression() {
+    return cronExpression;
+  }
+
+  public void setCronExpression(String cronExpression) {
+    this.cronExpression = cronExpression;
+  }
+
   public boolean updateTime() {
     if (new DateTime(nextExecTime).isAfterNow()) {
       return true;
     }
 
+    if (cronExpression != null) {
+      DateTime nextTime = getNextCronRuntime(nextExecTime, timezone, Utils.parseCronExpression(cronExpression, timezone));
```

Comment:
I believe this method will be called once when a schedule is created. It is intended to have the first Execution time, _**nextExecTime**_ in code. There exists come potential optimizations in these code. Will have deep look.